NebulaClient/PacketProcessors/Factory/Belt/BeltUpdatePickupItemsProcessor.cs

```diff
@@ -16,6 +16,10 @@ public void ProcessPacket(BeltUpdatePickupItemsPacket packet, NebulaConnection c
                 //Iterate though belt updates and remove target items
                 for (int i = 0; i < packet.BeltUpdates.Length; i++)
                 {
+                    if (packet.BeltUpdates[i].BeltId >= traffic.beltPool.Length)
```

Comment:
How would this happen? Shouldn't we just resize the beltpool?

Reply:
I am not exactly sure, but somebody posted a screenshot where this method throws `Index out of array`. It was when the belt was built, so I guess, it was built on the host side sooner than on the client-side and the ID in this packet was out of the array.

![](https://media.discordapp.net/attachments/832667834823213156/837741684544241764/unknown.png)

Reply:
That means it's a race condition bound to desync. Fine as-is for now, but dangerous long term.

Reply:
since client with high ping lives in "past", I am not sure if this can be prevented.